tsdb/store_test.go

```diff
@@ -2682,6 +2730,25 @@ func (s *Store) Reopen() error {
 	return s.Store.Open()
 }
 
+// Reopen closes and reopens the store as a new store.
+func (s *Store) ReopenWithStartupMetrics(msl *mockStartupLogger) error {
+	if err := s.Store.Close(); err != nil {
+		return err
+	}
+
+	s.Store = tsdb.NewStore(s.Path())
+	s.EngineOptions.IndexVersion = s.index
+	s.EngineOptions.Config.WALDir = filepath.Join(s.Path(), "wal")
+	s.EngineOptions.Config.TraceLoggingEnabled = true
+
+	s.WithStartupMetrics(msl)
+
+	if testing.Verbose() {
+		s.WithLogger(logger.New(os.Stdout))
+	}
+	return s.Store.Open()
+}
+
```

Comment:
I just refactored the test `Store.Open` and `Store.Reopen` to take options that either be persist when `Reopen` is called or changed when `Reopen` is called in `main-2.x`. You might take a look at them so we don't need a nearly identical copy and paste of `Reopen`.

* [main-2.x test Store.Open](https://github.com/influxdata/influxdb/blob/96bade409e73709bd01f9cadf1d745767887bfeb/tsdb/store_test.go#L2617-L2654)
* [main-2.x test Store.Reopen](https://github.com/influxdata/influxdb/blob/96bade409e73709bd01f9cadf1d745767887bfeb/tsdb/store_test.go#L2669-L2694)

Reply:
I need to take a look at this today 🫡